Jenkinsfile

```diff
@@ -0,0 +1,225 @@
+                        always {
+                            sh "docker cp lab-results-tests:/home/gradle/src/build ."
+                            junit '**/build/test-results/**/*.xml'
+                            step([
+                                $class : 'JacocoPublisher',
+                                execPattern : '**/build/jacoco/*.exec',
+                                classPattern : '**/build/classes/java',
+                                sourcePattern : 'src/main/java',
+                                exclusionPattern : '**/*Test.class'
+                            ])
+                            sh "rm -rf build"
+                        }
+                    }
+                }
+                stage('Build Docker Images') {
+                    steps {
+                        script {
+                            sh label: 'Running docker build', script: 'docker build -t ${DOCKER_IMAGE} .'
+                        }
+                    }
+                }
+                stage('Push image') {
+                    when {
+                        expression { currentBuild.resultIsBetterOrEqualTo('SUCCESS') }
+                    }
+                    steps {
+                        script {
+                            if (ecrLogin(TF_STATE_BUCKET_REGION) != 0 )  { error("Docker login to ECR failed") }
+                            String dockerPushCommand = "docker push ${DOCKER_IMAGE}"
+                            // Change echo to be an error once the AWS has been set up
+                            if (sh (label: "Pushing image", script: dockerPushCommand, returnStatus: true) !=0) { echo("Docker push image failed") }
+                        }
+                    }
+                }
+            }
+            post {
+                always {
+                    sh label: 'Copy lab-results container logs', script: 'docker-compose logs lab-results > logs/lab-results.log'
+                    sh label: 'Copy activemq logs', script: 'docker-compose logs activemq > logs/inbound.log'
+                    archiveArtifacts artifacts: 'logs/*.log', fingerprint: true
+                    sh label: 'Stopping containers', script: 'docker-compose down -v'
+                }
+            }
+        }
+        stage('Deploy and Integration Test') {
+            when {
+                expression { currentBuild.resultIsBetterOrEqualTo('SUCCESS') }
+            }
+            options {
+              lock("${tfProject}-${tfEnvironment}-${tfComponent}")
+            }
+            stages {
+                stage('Deploy using Terraform') {
+                    steps {
+                        script {
+                            String tfCodeBranch  = "develop"
+                            String tfCodeRepo    = "https://github.com/nhsconnect/integration-adaptors"
+                            String tfRegion      = "${TF_STATE_BUCKET_REGION}"
+                            List<String> tfParams = []
```

Comment:
I don't think this is used anywhere?

Reply:
@n-r-b well spotted! Has been removed now